packages/cli/src/cmds/preview/preview.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import * as path from "path";
+import { Argv } from "yargs";
+import { YargsCommand } from "../../yargsCommand";
+import { err, FxError, ok, Result } from "@microsoft/teamsfx-api";
+
+import * as utils from "../../utils";
+import * as commonUtils from "./commonUtils";
+import * as constants from "./constants";
+import { CliTelemetry } from "../../telemetry/cliTelemetry";
+import { WorkspaceNotSupported } from "./errors";
+
+export default class Preview extends YargsCommand {
+    public readonly commandHead = `preview`;
+    public readonly command = `${this.commandHead}`;
+    public readonly description = "Preview the current application.";
+
+    public builder(yargs: Argv): Argv<any> {
+        yargs.option("local", {
+            description: "Preview the current application from local",
+            boolean: true,
+            default: false,
+        });
+        yargs.option("remote", {
+            description: "Preview the current application from remote",
+            boolean: true,
+            default: false,
+        });
+        yargs.option("folder", {
+            description: "Select root folder of the project",
+            string: true,
+            default: "./",
+        })
+
+        return yargs.version(false);
+    }
+
+    public async runCommand(args: { [argName: string]: boolean | string | string[] | undefined }): Promise<Result<null, FxError>> {
+        const workspaceFolder = path.resolve(args.folder as string);
+        if (!utils.isWorkspaceSupported(workspaceFolder)) {
+            return err(WorkspaceNotSupported(workspaceFolder));
+        }
+
+        CliTelemetry.setReporter(CliTelemetry.getReporter().withRootFolder(workspaceFolder));
+
+        if (args.local || (!args.local && !args.remote)) {
```

Comment:
what if both `local` and `remote` are set to true? better to handle such case as user error

Reply:
Current behavior regards it as `local`. I think command like `teamsfx preview --type local` is better than `teamsfx preview --local`. However, since we have made the decision, let me just handle the exclusion case.

Reply:
Added. Also updated the option description.